prov/usnic/src/usdf_ep_dgram.c

```diff
@@ -157,13 +157,26 @@ usdf_ep_dgram_enable(struct fid_ep *fep)
 static int
 usdf_ep_dgram_bind(struct fid *fid, struct fid *bfid, uint64_t flags)
 {
+	int ret;
 	struct usdf_ep *ep;
 	struct usdf_cq *cq;
 	struct usdf_av *av;
-	int ret;
 
 	USDF_TRACE_SYS(EP_CTRL, "\n");
 
+	/* Backward compatibility case for Open MPI. We haven't been validating the flags until now.
+	 * Older version of Open MPI gives FI_RECV as AV bind flag (bug). */
+	if (bfid->fclass == FI_CLASS_AV) {
+		av = av_fidtou(bfid);
+		if (av->av_domain->dom_info->fabric_attr->api_version <= FI_VERSION(1, 4) && flags == FI_RECV)
```

Comment:
It might be slightly safer to check for `(flags & FI_RECV) != 0` (also in the RDM case, below).